source/Ship.cpp

```diff
@@ -1392,13 +1393,20 @@ vector<string> Ship::FlightCheck() const
 			checks.emplace_back("solar power?");
 		if(fuel < 0.)
 			checks.emplace_back("fuel?");
-		if(!canBeCarried)
+		if(canBeCarried && !intrasolar)
 		{
 			if(!hyperDrive && !jumpDrive)
 				checks.emplace_back("no hyperdrive?");
 			if(fuelCapacity < navigation.JumpFuel())
 				checks.emplace_back("no fuel?");
 		}
+		if(!canBeCarried)
+		{
+			if(baseAttributes.Category() != "Intrasolar" && (!hyperDrive && !jumpDrive))
+				checks.emplace_back("no hyperdrive?");
+			if(fuelCapacity < navigation.JumpFuel())
+				checks.emplace_back("no fuel?");
+		}
```

Comment:
```suggestion
		if(hyperDrive || jumpDrive)
		{
			if(fuelCapacity < navigation.JumpFuel())
				checks.emplace_back("no fuel?");
		}
```

This might be more to the point, skipping asking whether or not the ship can be carried or is intrasolar or anything, and just ask "does it have a drive?" If it does, check fuel. If not, don't.